src/main/java/it/gov/pagopa/reward/repository/UserInitiativeCountersAtomicOpsRepositoryImpl.java

```diff
@@ -0,0 +1,60 @@
+import it.gov.pagopa.reward.dto.synchronous.SynchronousTransactionRequestDTO;
+import it.gov.pagopa.reward.exception.TransactionSynchronousException;
+import it.gov.pagopa.reward.model.counters.UserInitiativeCounters;
+import it.gov.pagopa.reward.utils.RewardConstants;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.data.mongodb.core.ReactiveMongoTemplate;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.data.mongodb.core.query.Update;
+import reactor.core.publisher.Mono;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+public class UserInitiativeCountersAtomicOpsRepositoryImpl implements UserInitiativeCountersAtomicOpsRepository{
+    private final int throttlingSeconds;
+    private final ReactiveMongoTemplate mongoTemplate;
+    private final SynchronousTransactionRequestDTOt2TrxDtoOrResponseMapper synchronousTransactionRequestDTOt2TrxDtoOrResponseMapper;
+
+    public UserInitiativeCountersAtomicOpsRepositoryImpl(@Value("${app.synchronous.throttlingSeconds}") int throttlingSeconds, ReactiveMongoTemplate mongoTemplate, SynchronousTransactionRequestDTOt2TrxDtoOrResponseMapper synchronousTransactionRequestDTOt2TrxDtoOrResponseMapper) {
+        this.throttlingSeconds = throttlingSeconds;
+        this.mongoTemplate = mongoTemplate;
+        this.synchronousTransactionRequestDTOt2TrxDtoOrResponseMapper = synchronousTransactionRequestDTOt2TrxDtoOrResponseMapper;
+    }
+
+    @Override
+    public Mono<UserInitiativeCounters> updateDate(SynchronousTransactionRequestDTO request, String initiativeId) {
+        Mono<UserInitiativeCounters> initiativeCountersMono = mongoTemplate
+                .findAndModify(
+                        Query.query(criteriaByUserIdAndInitiativeId(request.getUserId(), initiativeId)
+                                .orOperator(
+                                        Criteria.where(UserInitiativeCounters.Fields.updateDate).is(null),
+                                        Criteria.where(UserInitiativeCounters.Fields.updateDate).is(LocalDateTime.now().minusSeconds(throttlingSeconds)))),
+                        new Update()
+                                .setOnInsert(UserInitiativeCounters.Fields.updateDate, LocalDateTime.now()),
+                        UserInitiativeCounters.class
+                );
+
+        initiativeCountersMono.hasElement().filter(b ->b.equals(Boolean.FALSE))
+                .flatMap(b -> mongoTemplate.exists(Query.query(criteriaByUserIdAndInitiativeId(request.getUserId(), initiativeId)), UserInitiativeCounters.class))
+                .map(b -> {
+                    if(b.equals(Boolean.TRUE)){
+                        throw new TransactionSynchronousException(synchronousTransactionRequestDTOt2TrxDtoOrResponseMapper.apply(request, initiativeId, List.of(RewardConstants.TRX_TOO_REJECTION_TOO_MANY_REQUEST)));
```

Comment:
the 429 will not have a body! so it's not necessary to invoke the mapper and throw the TransactionSynchronousException, just throw ClientExceptionNoBody